bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-notifications.js

```diff
@@ -89,12 +90,417 @@ export default function defineOHBlocks_Notifications (f7, isGraalJs) {
+        .appendField(tagImageField, 'tagIcon')
+        .appendField(referenceImageField, 'tagIcon')
+        .appendField(mediaImageField, 'mediaIcon')
+        .appendField(actionImageField, 'tagIcon')
+        .appendField(actionButton1ImageField, 'actionButton1Icon')
+        .appendField(actionButton2ImageField, 'actionButton2Icon')
+        .appendField(actionButton3ImageField, 'actionButton3Icon')
+        .appendField(new Blockly.FieldDropdown([['send notification', 'cloud'], ['send log', 'log']]), 'type')
+
+      this.updateShape(this.hasUser, true, this.hasIconInfo, this.hasTag, this.hasHeaderTitle, this.hasReferenceId, this.hasAction, this.hasActionButton1, this.hasMedia)
+
+      this.setPreviousStatement(true, null)
+      this.setNextStatement(true, null)
+      this.setInputsInline(false)
+      this.setColour(0)
+      this.setTooltip('Sends a notification to the cloud log only, not to any device (requires openHAB Cloud Connector)')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-notifications.html#sendCloudNotification')
+    },
+    updateShape: function (hasUser, addTextBlock, hasIconInfo, hasTag, hasHeaderTitle, hasReferenceId, hasAction, hasActionButton1, hasActionButton2, hasActionButton3, hasMedia) {
+      this.hasUser = hasUser
+      this.hasIconInfo = hasIconInfo
+      this.hasTag = hasTag
+      this.hasHeaderTitle = hasHeaderTitle
+      this.hasReferenceId = hasReferenceId
+      this.hasMedia = hasMedia
+      this.hasAction = hasAction
+      this.hasActionButton1 = hasActionButton1
+      this.hasActionButton2 = hasActionButton2
+      this.hasActionButton3 = hasActionButton3
+
+      const blockSequence = ['usersInput', 'headerTitle', 'icon', 'tag', 'reference', 'media', 'action', 'actionButton1', 'actionButton2', 'actionButton3']
+
+      /*
+       * adds a "shadow" text input to the parent block
+       */
+      function addBlockInput (parentConnection, tooltip, blockType = 'text') {
+        const textUsersBlock = this.workspace.newBlock(blockType)
+        textUsersBlock.setTooltip(tooltip)
+        textUsersBlock.initSvg()
+        textUsersBlock.render()
+        parentConnection.connect(textUsersBlock.outputConnection)
+      }
+
+      /*
+       * removes the block "shadow" text input from the block
+       */
+      function removeTextInput (parentConnection) {
+        const targetBlock = parentConnection.targetBlock()
+        if (targetBlock) {
+          targetBlock.unplug(true)
+          targetBlock.dispose(true)
+        }
+      }
+
+      /*
+       * show or hide this particular block
+       * toggleOn : show (true) / hide (false) block
+       * blockName: name of the block to be toggled
+       * label: label for the preceding label block
+       * tooltip: tooltip for the input block
+       */
+      function toogleBlock (toggleOn, blockName, label, tooltip, blockTypeCheck = 'String', shadowBlock = 'text') {
```

Comment:
FYI if you make this a proper JSDoc comment, your IDE can support you better.
Once you are finished with this PR, I will add a commit to make this JSDoc.

Reply:
I added JSDoc

Reply:
Hmm it was partly broken, I have fixed it.